buzzmobile/tests/rostest_utils.py

```diff
@@ -0,0 +1,161 @@
+import roslaunch
+import functools
+import subprocess
+import rospy
+import contextlib
+import rosnode
+import time
+
+def with_roscore(obj):
+    """Decorator to run all tests in a testcase with their own roscore.
+
+    This wraps the setUp and tearDown methods to start by first spinning up a
+    roscore process, and tears it down at the very end. This adds a small time
+    penalty, but its worth it.
+
+    Its worth deciding whether to make this work on a per-method, per object,
+    or both basis.
+    """
+    old_setup = obj.setUp
+    old_teardown = obj.tearDown
+
+    def new_setup(self):
+        self.roscore = subprocess.Popen(['roscore'])
+        old_setup(self)
+
+    def new_teardown(self): 
+        old_teardown(self)
+        self.roscore.kill()
+        subprocess.call(['killall', '-9', 'rosmaster'])
+        self.roscore.wait()
+
+    obj.setUp = new_setup
+    obj.tearDown = new_teardown
+    return obj
+
+
+class ROSLauncher(roslaunch.scriptapi.ROSLaunch):
+    """
+    ROSLaunch but allows the use of launch files.
+
+    This was found by peering into the roslaunch util source code from
+    `which roslaunch`. 
+    """
+    def __init__(self, files):
+        super().__init__()
+        uuid = roslaunch.rlutil.get_or_generate_uuid(None, True) 
+        self.parent = roslaunch.parent.ROSLaunchParent(uuid, files, is_core=False) 
+
+# TODO(joshuamorton@gatech.edu): swap this to a Map[int, launcher], where the
+# int is incrementally generated and stored within the calling decorator. This
+# will prevent the issue of overwriting or cross writing if it becomes a
+# danger.
```

Comment:
If we have multiple tests, does pytest not parallelize things? (And therefore lead to problems?)